File: Исходники/Глава 2. Часть 1/Динамическое программирование1/Динамическое программирование1/Табличный метод/24. Бином Ньютона. Формула/C++/Олимпиадный/binom_formula.cpp
#include <bits/stdc++.h>
#include <cassert>

using namespace std;

int fact(int n) {
    if (n == 1) {
        return 1;
    }
    int answer = 1;
    for (int i = 2; i < n + 1; ++i) {
        answer *= i;
    }
    return answer;
}

int combination(int n, int k) {
    if (n <= 0 or k <= 0 or k > n) {
        return -1;
    }
    int numer = fact(n);
    int denom = fact(n - k) * fact(k);
    int answer = numer / denom;
    return answer;
}

File: Исходники/Глава 2. Часть 1/Динамическое программирование1/Динамическое программирование1/Табличный метод/24. Бином Ньютона. Формула/Python/binom_formula.py
def fact(n):
    if n == 1:
        return 1
    answer = 1
    for i in range(2, n + 1):
        answer *= i
    return answer


def combinations(n, k):
    if n <= 0 or k <= 0 or k > n:
        return -1
    numer = fact(n)
    denom = fact(n - k) * fact(k)
    answer = numer // denom
    return answer

